server/server.js

```diff
@@ -1,15 +1,125 @@
 const express = require("express");
+const cors = require("cors");
 const app = express();
-const port = process.env.PORT || 5000;
+const port = process.env.PORT || 5050;
+const { Pool } = require("pg");
+require("dotenv").config();
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+const pool = new Pool({
+  connectionString: process.env.DB_URL,
+  ssl: { rejectUnauthorized: false },
+});
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+app.use(cors());
+app.use(express.json());
 
-// GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+//  Get all of the videos
+app.get("/videos", async (req, res) => {
+  try {
+    const query = "SELECT * FROM videos";
+    const result = await pool.query(query);
+    const videos = result.rows;
+    res.status(200).json(videos);
+  } catch (error) {
+    console.error(error);
+  }
+});
+// let idCounter = Math.max(...videos.map((video) => video.id)) + 1;
+
+//  Add a video to the API.
+app.post("/videos", async (req, res) => {
+  try {
+    //  if  (!req.body.title || !req.body.url) {
+    //     return res.status(400).json({ Message: "Video could not be saved" });
+    //   }
+    const { title, url } = req.body;
+    const rating = 0;
+    const query =
+      "INSERT INTO videos (title, url, rating) VALUES ($1, $2, $3) RETURNING *";
+    const values = [title, url, rating];
+    const result = await pool.query(query, values);
+    const newVideo = result.rows[0];
+    res.status(201).json(newVideo);
+  } catch (error) {
+    console.error(error);
+  }
+  // const newVideos = {
+  //   id: idCounter,
+  //   title: req.body.title,
+  //   url: req.body.url,
+  //   rating: 0,
+  // };
+  // videos.push(newVideos);
+  // console.log("New video added:", newVideos);
+  // res.json({ id: newVideos.id });
```

Comment:
Theres lots of commented code around, would be good to clean this up, it makes it harer to review 😄